components/contexts/Web3Context.js

```diff
@@ -24,12 +19,13 @@ const providerOptions = {
 
 export const Web3Provider = ({ children }) => {
   const [web3Modal, setWeb3Modal] = useState()
-  const [accounts, setAccounts] = useState([])
+  const [signer, setSigner] = useState()
+  const [address, setAddress] = useState()
   const [contracts, setContracts] = useState({})
   const [web3, setWeb3] = useState()
   const [provider, setProvider] = useState()
   const [networkId, setNetworkId] = useState()
-  const [error, setError] = useState()
+  const [networkError, setNetworkError] = useState()
```

Comment:
Setting errors more explicitly so that components can compose how they handle specific errors.

Reply:
💯